### PR TITLE
initial commit Aligned Button and title

### LINKED ISSUE #252: Aligned all elements on user cards
## Description of ideal solution

can you align the buttons, icons, and title on the user cards 
## Alternative solutions if any [optional]
- buttons and titles on user cards should be well-aligned marginLeft of buttons should be the same as the titles.
- Make sure you also same left margin in the People/Users cards buttons and headings.
- The marginLeft can be 16px.
- Make sure that the assignees, and reviewers email addresses are also aligned with the buttons and titles.

## Related issues

[Align-Tittle-and-buttons-in-users-cards](https://github.com/Umuzi-org/Tilde/issues/229)

### PR DESCRIPTION
**Pull request by kaleem99**
Related issues: [please specify]

## Description:

What are you up to? Fill us in :)

## I solemnly swear that:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by kaleem99** [comment]
align-title

**Comment by kaleem99** [comment]
aligned-title2

**Review by babalwa-mbolekwa — CHANGES_REQUESTED** [review]
Hey Kaleem could you please double check if you have imported correctly from material UI. I got this errors on the frontend
errors

**Review by Vuyisanani — CHANGES_REQUESTED** [review]
Hi @kaleem99 

Great work so far, just one small thing for consistency, can you add a margin/padding to the titles here as well, just to align them with the email addresses.

Screenshot 2021-11-16 at 09 46 42


Screenshot 2021-11-16 at 09 42 45

**Comment by kaleem99** [comment]
align

**Review by Vuyisanani — APPROVED** [review]

File: frontend/src/components/widgets/BaseButtonLink.js [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
-}
\ No newline at end of file
+import { makeStyles } from "@material-ui/core/styles";
+import { Button } from "@material-ui/core";
+export const useStyles = makeStyles({
+  title: {
+    marginTop: "8px",
+    marginLeft: "16px",
+  },
+  input: {
```
Comment (sheenarbw) [inline frontend/src/components/widgets/BaseButtonLink.js]: I'm not sure what input is doe. You don't seem to use it.

And why is the other class named title? That's an odd name for a button's class.

And what happens if we put the button underneath another element that hes marginBottom set? now other elements needs to be aware of the margins set up in this file. I'm not sure it makes sense to have the classes here at all.

File: frontend/src/components/regions/UserDetails/TeamsTable/Presentation.js [hunk 2]
```diff
@@ -28,7 +28,7 @@ const TeamsTable = ({ teams, authUser }) => {
             className={classes.tableHead}
             colSpan="2"
           >
-            <Typography variant="h6" component="h2">
+            <Typography variant="h6" component="h2" style={{ width: "90%" }}>
```
Comment (sheenarbw) [inline frontend/src/components/regions/UserDetails/TeamsTable/Presentation.js]: why do we sometimes use classes and sometimes use inline styles?

**Review by sheenarbw — CHANGES_REQUESTED** [review]
Looks good but there are a few little things

File: frontend/src/components/widgets/LinkToUserBoard.js [hunk 3]
```diff
@@ -1,18 +1,20 @@
 import React from "react";
 import { routes } from "../../routes";
-
 import BaseButtonLink from "./BaseButtonLink";
+import { useStyles } from "./BaseButtonLink";
```
Comment (sheenarbw) [inline frontend/src/components/widgets/LinkToUserBoard.js]: Please rather style things in place. Spagetti styling makes things hard to reason about. 

Importing and exporting styles from all over the place makes it hard to know what things do.

If styles are used in 1 component put them in that component.

Global styles should not be defined in strange littel components.

**Review by sheenarbw — CHANGES_REQUESTED** [review]
Some changes weren't addressed before. 

And styles are quite spegitti like. Please look into KISS and the principle of least privelage. DRY code should never ever imply spagetti code

**Review by babalwa-mbolekwa — APPROVED** [review]

**Review by Sbonelo01 — APPROVED** [review]

File: frontend/src/components/widgets/BaseButtonLink.js (hunk 1 above)
Comment (sheenarbw) [reply frontend/src/components/widgets/BaseButtonLink.js]: Bump.

Do you use `input` anywhere? I can't see it being used. 
We should try not to add extra bloat to the code. If something isn't used then it shouldn't be there.

File: frontend/src/components/widgets/LinkToUserBoard.js [hunk 4]
```diff
@@ -1,18 +1,27 @@
...
-
 import BaseButtonLink from "./BaseButtonLink";
-
+import { makeStyles } from "@material-ui/core/styles";
+const useStyles = makeStyles({
+  marginsAlignment: {
+    marginTop: "8px",
+    marginLeft: "16px",
+  },
+  input: {
```
Comment (sheenarbw) [inline frontend/src/components/widgets/LinkToUserBoard.js]: Is input being used here?

**Review by sheenarbw — CHANGES_REQUESTED** [review]
There's a little bloat going on.

If we have a big team of devs and every single person were to add random pieces of unused code in all their PRs then our cde base would become very confusig and bloated. It pays to be tidy about things.

Can you please remove the code that doesn't do anything?

**Review by Vuyisanani — APPROVED** [review]

File: frontend/src/components/regions/UsersAndGroups/Presentation.js [hunk 5]
```diff
@@ -156,14 +182,19 @@ export default function Presentation({
...
+            className={classes.marginsAlignment}
+          >
             Users
           </Typography>
           <TextField
             label={`${filterUsersByGroupName} Users`}
             variant="outlined"
             {...filterByUser}
-            className={classes.textBoxSize}
+            className={classes.input}
```
Comment (Vuyisanani) [inline frontend/src/components/regions/UsersAndGroups/Presentation.js]: Why are you using input class on a TextField?

**Review by Vuyisanani — CHANGES_REQUESTED** [review]

**Comment by Vuyisanani** [comment]
You can use the same input, maybe rename the class to a better name for that TextField and use the exact properties

**Review by babalwa-mbolekwa — APPROVED** [review]

**Review by Vuyisanani — APPROVED** [review]

**Review by sheenarbw — APPROVED** [review]

**Pull request merged**